src/test/java/net/kyori/text/LegacyComponentTest.java

```diff
@@ -38,6 +38,6 @@ public void testFrom() {
   @Test
   public void testFromColor() {
     assertEquals(TextComponent.of("foo").color(TextColor.GREEN), LegacyComponent.from(LegacyComponent.CHARACTER + "afoo"));
```

Comment:
To be fair, this test relies exactly on the one case where you could make the optimization, so you either need to add it back for `pos == 0 && parts == 1` or modify what's expected.